src/deploy.rs

```diff
@@ -23,25 +22,61 @@ pub struct Cmd {
 
 #[derive(thiserror::Error, Debug)]
 pub enum Error {
-    #[error("io")]
-    Io(#[from] io::Error),
-    #[error("xdr")]
+    #[error(transparent)]
+    // TODO: the Display impl of host errors is pretty user-unfriendly
+    //       (it just calls Debug). I think we can do better than that
+    Host(#[from] HostError),
```

Comment:
Instead of fixing the host error here, we should fix this in the host, so that we address the problem in all the places it's going to show up (here in the CLI, in Rust tests of contracts, etc). cc @graydon @jayz22
```suggestion
    #[error(transparent)]
    Host(#[from] HostError),
```